File: Problem7/problem7.java
class problem7
{
    public static void main(String args[])
    {
        int n=2,i=0;
        while(i<10001)
        {
        if(isprime(n))
        {
            i++;
        }
        n+=1;
    }
    System.out.println("10001st prime number="+(n-1));
}
public static boolean isprime(int n)
{
    if(n==2)
    return true;
    else if(n % 2==0)
    return false;
    else{
    double s=Math.sqrt(n);
    
    for(int c=3;c<=s;c+=2)
    {
        if(n%c==0)
        return false;
    }
    return true;
}
}
}

        
        
            

File: Problem7/problem7.py
def main():
	n=2
	i=0
	while(i<10001):
		if(isprime(n)):
			i=i+1
		n+=1
	print("10001st prime no=",n-1)

def isprime(n):
	if(n==2):
		return True
	elif(n%2==0):
		return False
	else:
		s=int(n**0.5)+1

		for c in range(3,s,2):
			if(n%c==0):
				return False
		return True

main()